syncd/scripts/syncd_init_common.sh

```diff
@@ -122,74 +122,188 @@ config_syncd_cisco_8000()
+                echo "Merge properties with override $override"
+            else
+                sedline=${line%:*}
+                if grep -q $sedline $to_file ;then
+                   if $override ;then
+                      echo "Override the config $(grep $sedline $to_file) with $line in $to_file"
+                      prop=${line:0:`expr index "$line" :`}
+                      sed -i "/$prop/d" $to_file
+                      echo "        $line" >> $to_file
+                      merged_cnt+=1
                    else
-                       sedline=${line%=*}
-                       if grep -q $sedline $file ;then
-                          echo "Keep the config $(grep $sedline $file) in $file"
-                       else
-                          echo $line >> $file
-                       fi
+                      grepline=$(grep $sedline $to_file)
+                      grepline="${grepline#"${grepline%%[![:space:]]*}"}"
+                      if [ "${grepline::1}" == '#' ];then
+                         echo "        $line" >> $to_file
+                         merged_cnt+=1
+                      else
+                         echo "Keep the config $(grep $sedline $to_file) in $to_file"
+                      fi
                    fi
-               fi
-             done < $COMMON_CONFIG_BCM
-             echo "# End of chip common properties" >> $file
-         done
-         echo "Merging $PLT_CONFIG_BCM with $COMMON_CONFIG_BCM, merge files stored in $CONFIG_BCM"
-      fi
-
-      #sync the file system
-      sync
-
-      # copy the final config.bcm and sai.profile to the shared folder for 'show tech'
-      cp -f /tmp/sai.profile /var/run/sswsyncd/
-      cp -f /tmp/*.bcm /var/run/sswsyncd/
-
-      if [ -f "/tmp/sai.profile" ]; then
-          CMD_ARGS+=" -p /tmp/sai.profile"
-      elif [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+                else
+                   echo "        $line" >> $to_file
+                   merged_cnt+=1
+                fi
+            fi
+        fi
+    done < $from_file
 
-    else
+    if [ $merged_cnt -gt 0 ]; then
+         sed -i "/# Start of/a \    global:" $to_file
+         sed -i "/# Start of/a \  0:" $to_file
+         sed -i "/# Start of/a \bcm_device:" $to_file
+         sed -i "/# Start of/a \---" $to_file
+    fi
+    echo "# End of $message" >> $to_file
+    if [ $merged_cnt -gt 0 ]; then
+       sed -i "/# End of/i \..." $to_file
+    fi
+    echo "Merged $from_file to $to_file"
+}
 
-      if [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+config_syncd_bcm()
+{
+    PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+    PLT_CONFIG_BCM=""
+    PLT_CONFIG_YML=""
+    PLT_SAI_PROFILE=$(find $HWSKU_DIR -name 'sai.profile')
```

Comment:
@geans-pin sai.profile should located in `/etc/sai.d` folder
Take 6100 for example, sai.profile is not in `/usr/share/sonic/hwsku`


```
root@str3-s6100-acs-7:/etc/sai.d# cat sai.profile 
SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/th-s6100-64x40G-t1.config.bcm
SAI_NUM_ECMP_MEMBERS=64
root@str3-s6100-acs-7:/etc/sai.d# cd /usr/share/sonic/hwsku
root@str3-s6100-acs-7:/usr/share/sonic/hwsku# ls -al
total 41
drwxr-xr-x 5 root root   384 Mar 14  2024 .
drwxr-xr-x 1 root root  4096 Nov 27 02:49 ..
drwxr-xr-x 2 root root   123 Mar 14  2024 BALANCED
drwxr-xr-x 2 root root   123 Mar 14  2024 RDMA-CENTRIC
drwxr-xr-x 2 root root   123 Mar 14  2024 TCP-CENTRIC
-rw-r--r-- 1 root root   221 Feb 20  2017 buffer_ports_t0.j2
-rw-r--r-- 1 root root   221 Feb 20  2017 buffer_ports_t1.j2
-rw-r--r-- 1 root root    66 Feb 20  2017 buffers.json.j2
-rw-r--r-- 1 root root  1238 Feb 20  2017 buffers_defaults_t0.j2
-rw-r--r-- 1 root root  1238 Feb 20  2017 buffers_defaults_t1.j2
-rw-r--r-- 1 root root   812 Feb 20  2017 pg_profile_lookup.ini
-rw-r--r-- 1 root root  3953 Feb 20  2017 port_config.ini
-rw-r--r-- 1 root root   775 Feb 20  2017 qos.json.j2
-rw-r--r-- 1 root root   728 Feb 20  2017 sai.profile.j2
-rw-r--r-- 1 root root 10607 Feb 20  2017 th-s6100-64x40G-t0.config.bcm
-rw-r--r-- 1 root root 10650 Feb 20  2017 th-s6100-64x40G-t1.config.bcm
```

Reply:
I've changed the search path to /etc/sai.d  which should be common for all broadcom sonic platforms